build.gradle

```diff
@@ -118,6 +120,24 @@ subprojects {
         assertsEnabled = true
       }
     }
+
+    options.compilerArgs += [
+        // enable all warnings
+        '-Xlint:all',
+        // hide warning: No processor claimed any of these annotations: dagger.*,org.junit.Test
+        '-Xlint:-processing',
+        // hide warning: MethodParameters attribute introduced in version 52.0 class files is ignored in version 51.0 class files
+        '-Xlint:-classfile',
+        // hide warning: auxiliary class C in full/path/to/C.java should not be accessed from outside its own source file
+        '-Xlint:-auxiliaryclass'
+    ]
+
+    if (project.property("strict.compilation").toBoolean()) {
```

Comment:
Can we do something like `findProperty("string.compilation") == "true"` to avoid having to change `gradle.properties`?

Reply:
Sure, if you prefer that. The reason behind a default value in gradle.properties was to be explicit; cleanly tell: "this key exists" and "this is the default value". This way multiple usages of the same property can be defaulted by using Gradle's features, instead of potentially duplicating code, just used to doing this nowadays.